cloudinit/sources/DataSourceVMware.py

```diff
@@ -776,91 +774,52 @@ def get_default_ip_addrs():
     addresses associated with the device used by the default route for a given
     address.
     """
-    # TODO(promote and use netifaces in cloudinit.net* modules)
-    gateways = netifaces.gateways()
-    if "default" not in gateways:
-        return None, None
-
-    default_gw = gateways["default"]
-    if (
-        netifaces.AF_INET not in default_gw
-        and netifaces.AF_INET6 not in default_gw
-    ):
-        return None, None
-
     ipv4 = None
     ipv6 = None
+    routes = netinfo.route_info()
+    for route in routes:
+        if "G" in route["flags"]:
+            if ipv4 and ipv4 != route["gateway"]:
+                LOG.debug(
+                    "multiple ipv4 gateways: %s, %s", ipv4, route["gateway"]
+                )
+            ipv4 = route["gateway"]
```

Comment:
I think route["gateway"] should be a gateway address, so here ipv4 should be gw4. Furthermore, can this make sure this is a default gateway?